sdk/test/trace/always_on_sampler_test.cc

```diff
@@ -0,0 +1,43 @@
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+
+#include <gtest/gtest.h>
+#include <map>
+
+using namespace opentelemetry::sdk::trace;
+using namespace opentelemetry::nostd;
+
+TEST(AlwaysOnSampler, ShouldSample)
+{
+  AlwaysOnSampler sampler;
+
+  // A buffer of trace_id with no specific meaning
+  constexpr uint8_t buf[] = {0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2, 3, 4, 5, 6, 7};
```

Comment:
nit: you probably don't need such a large buffer for testing :)

Reply:
It has already been merged. I will make another PR when we do more thorough testing :)